src/transformers/models/t5/modeling_tf_t5.py

```diff
@@ -1302,29 +1284,33 @@ def call(
         loss = None if inputs["labels"] is None else self.compute_loss(inputs["labels"], logits)
 
         past = (
-            (encoder_outputs, decoder_outputs[1]) if cast_bool_to_primitive(use_cache, self.config.use_cache) else None
+            (inputs["encoder_outputs"], decoder_outputs[1])
+            if cast_bool_to_primitive(inputs["use_cache"], self.config.use_cache)
```

Comment:
We still need `cast_bool_to_primitive`? I kinda thought this PR can resolve the problem

Reply:
Indeed, but I wanted to make sure of that in a later PR that will focus only on the T5 issues.